inference-engine/src/mkldnn_plugin/nodes/detectionoutput.cpp

```diff
@@ -24,98 +29,85 @@ static bool SortScorePairDescend(const std::pair<float, T>& pair1,
-            _variance_encoded_in_target = layer->GetParamAsBool("variance_encoded_in_target", false);
-            _keep_top_k = layer->GetParamAsInt("keep_top_k", -1);
-            _nms_threshold = layer->GetParamAsFloat("nms_threshold");
-            _confidence_threshold = layer->GetParamAsFloat("confidence_threshold", -FLT_MAX);
-            _share_location = layer->GetParamAsBool("share_location", true);
-            _clip_before_nms = layer->GetParamAsBool("clip_before_nms", false) ||
-                               layer->GetParamAsBool("clip", false);  // for backward compatibility
-            _clip_after_nms = layer->GetParamAsBool("clip_after_nms", false);
-            _decrease_label_id = layer->GetParamAsBool("decrease_label_id", false);
-            _normalized = layer->GetParamAsBool("normalized", true);
-            _image_height = layer->GetParamAsInt("input_height", 1);
-            _image_width = layer->GetParamAsInt("input_width", 1);
+            std::string errorMessage;
+            if (!isSupportedOperation(op, errorMessage)) {
+                THROW_IE_EXCEPTION_WITH_STATUS(NOT_IMPLEMENTED) << errorMessage;
+            }
+            auto doOp = ngraph::as_type_ptr<const ngraph::op::v0::DetectionOutput>(op);
+            auto attributes = doOp->get_attrs();
+
+            _num_classes = attributes.num_classes;
+            _background_label_id = attributes.background_label_id;
+            _top_k = attributes.top_k;
+            _variance_encoded_in_target = attributes.variance_encoded_in_target;
+            _keep_top_k = attributes.keep_top_k[0];
+            _nms_threshold = attributes.nms_threshold;
+            _confidence_threshold = attributes.confidence_threshold;
+            _share_location = attributes.share_location;
+            _clip_before_nms = attributes.clip_before_nms;
+            _clip_after_nms = attributes.clip_after_nms;
+            _decrease_label_id = attributes.decrease_label_id;
+            _normalized = attributes.normalized;
+            _image_height = attributes.input_height;
+            _image_width = attributes.input_width;
             _prior_size = _normalized ? 4 : 5;
             _offset = _normalized ? 0 : 1;
             _num_loc_classes = _share_location ? 1 : _num_classes;
 
-            with_add_box_pred = layer->insData.size() == 5;
-            _objectness_score = layer->GetParamAsFloat("objectness_score", 0.0f);
+            with_add_box_pred = op->get_input_size() == 5;
+            _objectness_score = attributes.objectness_score;
 
-            std::string code_type_str = layer->GetParamAsString("code_type", "caffe.PriorBoxParameter.CORNER");
-            _code_type = (code_type_str == "caffe.PriorBoxParameter.CENTER_SIZE" ? CodeType::CENTER_SIZE
-                                                                                 : CodeType::CORNER);
+            _code_type = (details::CaselessEq<std::string>()(attributes.code_type, "caffe.PriorBoxParameter.CENTER_SIZE") ?
```

Comment:
We have to check that `_code_type` is from supported range inside `isSupportedOperation`.

Reply:
Done